pkg/cloudprovider/provider/azure/create_delete_resources.go

```diff
@@ -0,0 +1,282 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-04-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-04-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// deleteInterfacesByMachineUID will remove all network interfaces tagged with the specific machine's UID.
+// The machine has to be deleted or disassociated with the interfaces beforehand, since Azure won't allow
+// us to remove interfaces connected to a VM.
+func deleteInterfacesByMachineUID(ctx context.Context, c *config, machineUID types.UID) error {
+	ifClient, err := getInterfacesClient(c)
+	if err != nil {
+		return fmt.Errorf("failed to create interfaces client: %v", err)
+	}
+
+	list, err := ifClient.List(ctx, c.ResourceGroup)
+	if err != nil {
+		return fmt.Errorf("failed to list interfaces in resource group %q", c.ResourceGroup)
+	}
+
+	var allInterfaces []network.Interface
+
+	for {
```

Comment:
why not `for list.NotDone() {`?

Reply:
Because that would skip the last page.

Reply:
https://sourcegraph.com/github.com/kubermatic/machine-controller@91b500fd63c88454374cdd949b9a1b75ccaa509e/-/blob/vendor/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-04-01/network/models.go#L12852:59

Reply:
right. fixed